apps/orders/views.py

```diff
@@ -1,37 +1,83 @@
+from apps.users.models import Address
+from apps.inventory.models import Inventory
 
 
-class Orders(
-    mixins.UpdateModelMixin,
-    mixins.ListModelMixin,
-    mixins.RetrieveModelMixin,
-    GenericViewSet,
-):
-    queryset = Order.objects.all()
+class OrderViewSet(ModelViewSet):
     serializer_class = OrderSerializer
 
+    http_method_names = [
+        m for m in ModelViewSet.http_method_names if m not in ["put", "patch"]
+    ]
+
     def get_queryset(self):
         user = self.request.user
-        if user.is_superuser:
-            return self.queryset
-        return Order.objects.filter(user=user)
+        if user.is_authenticated:
+            return Order.objects.filter(user=user)
+        return Order.objects.none()
 
+    @action(detail=False, methods=["post"], url_path="checkout")
+    @transaction.atomic
+    def checkout(self, request):
+        cart = ShoppingCart.objects.filter(user=request.user).first()
+        if not cart:
+            return Response({"detail": "Cart is empty"}, status=status.HTTP_400_BAD_REQUEST)
+        cart_items = CartItem.objects.filter(cart=cart.id)
 
-class OrderItems(
-    mixins.UpdateModelMixin,
-    mixins.ListModelMixin,
-    mixins.RetrieveModelMixin,
-    GenericViewSet,
-):
-    queryset = OrderItem.objects.all()
-    serializer_class = OrderItemSerializer
+        total_amount = sum(item.product.price * item.quantity for item in cart_items)
+        delivery_cost = 10
+        user_address = Address.objects.filter(user=request.user).first()
 
-    def get_queryset(self):
-        user = self.request.user
-        if user.is_superuser:
-            return self.queryset
-        return OrderItem.objects.filter(order__user=user)
+        order = Order.objects.create(
+            user=request.user,
+            delivery_cost=delivery_cost,
+            total_cost=total_amount + delivery_cost,
+            delivery_address=user_address,
+            status='PE'
+        )
+
+        for cart_item in cart_items:
+            OrderItem.objects.create(
+                order=order,
+                product=cart_item.product,
+                quantity=cart_item.quantity,
+                price=cart_item.price,
+            )
+
+        serializer = self.get_serializer(order)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    @action(detail=True, methods=['post'])
+    @transaction.atomic
+    def process_payment(self, request, pk=None):
+        order = self.get_object()
+
+        success = True  # I simulated the payment gateway integration
+
+        if success:
+            Transaction.objects.create(
+                user=request.user,
+                order=order,
+                amount=order.total_cost,
+                status='PA',
+                payment_method="CD"
+            )
+
+            order.status = 'PC'
+            order.save()
+
+            ShoppingCart.objects.filter(user=request.user).delete()
```

Comment:
I think to can move this to the checkout flow since the order has already been created so there wont be a need for it again